conda-recipes/D47188-svml.patch

```diff
@@ -560,30 +533,41 @@ index 37ffac1..8ad2131 100644
      // This convention allows using the Win64 convention on other targets.
      case CallingConv::Win64:
 diff --git a/lib/Transforms/Vectorize/LoopVectorize.cpp b/lib/Transforms/Vectorize/LoopVectorize.cpp
-index 5bcf0c0..0c4f666 100644
+index 5bcf0c0..cef0009 100644
 --- a/lib/Transforms/Vectorize/LoopVectorize.cpp
 +++ b/lib/Transforms/Vectorize/LoopVectorize.cpp
-@@ -4917,6 +4917,7 @@ void InnerLoopVectorizer::widenInstruction(Instruction &I) {
+@@ -3974,6 +3974,12 @@ static unsigned getVectorCallCost(CallInst *CI, unsigned VF,
+   if (!TLI || !TLI->isFunctionVectorizable(FnName, VF) || CI->isNoBuiltin())
+     return Cost;
+ 
++  // this goes against LLVM coding philosophy, but it'll stop bleeding
++  bool IgnoreMe;
++  StringRef FuncName = TLI->getVectorizedFunction(FnName, VF, IgnoreMe, true);
++  if (FuncName.startswith("__svml") && !TTI.isTypeLegal(RetTy))
++    return Cost;
++
+   // If the corresponding vector cost is cheaper, return its cost.
+   unsigned VectorCallCost = TTI.getCallInstrCost(nullptr, RetTy, Tys);
+   if (VectorCallCost < Cost) {
+@@ -4917,6 +4923,7 @@ void InnerLoopVectorizer::widenInstruction(Instruction &I) {
        }
  
        Function *VectorF;
 +      bool FromSVML = false;
        if (UseVectorIntrinsic) {
          // Use vector version of the intrinsic.
          Type *TysForDecl[] = {CI->getType()};
-@@ -4925,7 +4926,10 @@ void InnerLoopVectorizer::widenInstruction(Instruction &I) {
+@@ -4925,7 +4932,8 @@ void InnerLoopVectorizer::widenInstruction(Instruction &I) {
          VectorF = Intrinsic::getDeclaration(M, ID, TysForDecl);
        } else {
          // Use vector version of the library call.
 -        StringRef VFnName = TLI->getVectorizedFunction(FnName, VF);
-+        // TEMPORARY. For backporting (down to release 5.0.0) purposes let's use any() from FastMathFlags instead of 
-+        // still absent isFast(). It isn't equivalent for full mask but I think it's enough there
-+        bool IsFast = CI->getFastMathFlags().any();
++        bool IsFast = CI->getFastMathFlags().isFast();
```

Comment:
hmm.. I havn't noticed that before, is it what you implemented @stuartarchibald ? Did you need it for LLVM 6.0.0? I havn't noticed any issue with `.isFast()`

Reply:
I guess it is up to the difference between earlier version and published version of the patch

Reply:
Think it was the latter, I don't recall writing that :)